src/state/slices/portfolioSlice/portfolioSlice.test.ts

```diff
@@ -19,456 +38,746 @@ import {
-  }
-}
 
-const state = {
-  ...mockStore,
-  assets: {
-    byId: {
-      [ethCaip19]: ethereum,
-      [foxCaip19]: fox
-    },
-    ids: [ethCaip19, foxCaip19]
-  },
-  marketData: {
-    ...mockStore.marketData,
-    byId: {
-      [ethCaip19]: {
-        price: '1000',
-        marketCap: '10000',
-        volume: '100000',
-        changePercent24Hr: 10
-      },
-      [foxCaip19]: {
-        price: '1',
-        marketCap: '10000',
-        volume: '100000',
-        changePercent24Hr: 10
-      }
-    },
-    ids: [ethCaip19]
-  },
-  portfolio: {
-    ...mockStore.portfolio,
-    assetBalances: {
-      byId: {
-        [ethCaip19]: '115607633096574740',
-        [foxCaip19]: '105458486654698803892'
-      },
-      ids: [ethCaip19, foxCaip19]
-    },
-    accountBalances: {
-      byId: {
-        [ethAccountSpecifier1]: {
-          [ethCaip19]: '27803816548287370',
-          [foxCaip19]: '42729243327349401946'
-        },
-        [ethAccountSpecifier2]: {
-          [ethCaip19]: '87803816548287370',
-          [foxCaip19]: '62729243327349401946'
-        }
-      },
-      ids: [ethAccountSpecifier1, ethAccountSpecifier2]
-    }
-  }
-}
+      const ethAccountId = `${ethAccount.caip2}:${toLower(ethAccount.pubkey)}`
+      const ethAccount2Id = `${ethAccount2.caip2}:${toLower(ethAccount2.pubkey)}`
 
-describe('accountToPortfolio', () => {
-  it('can normalize eth and btc accounts to portfolio', () => {
-    const accounts = {
-      [ethAccount1.pubkey]: ethAccount1,
-      [ethAccount2.pubkey]: ethAccount2,
-      [btcAccount.pubkey]: btcAccount
-    }
-    const result = accountToPortfolio(accounts)
-    expect(result).toEqual(portfolio)
+      const state = store.getState()
+
+      const selected = selectPortfolioAssetAccounts(state, ethCaip19)
+      const expected = [ethAccountId, ethAccount2Id]
+      expect(selected).toEqual(expected)
+    })
   })
-})
 
-describe('selectPortfolioAssetAccounts', () => {
-  it('can get accounts containing an asset', () => {
-    const fooAccount = '0xfoo'
-    const barAccount = '0xbar'
-    const bazAccount = '0xbaz'
-
-    const state = {
-      ...mockStore,
-      portfolio: {
-        ...mockStore.portfolio,
-        accounts: {
-          byId: {
-            [fooAccount]: [ethCaip19],
-            [barAccount]: [ethCaip19],
-            [bazAccount]: []
-          },
-          ids: [fooAccount, barAccount, bazAccount]
-        }
-      }
-    }
+  describe('selectAccountIdByAddress', () => {
+    const store = createStore()
+    const { ethAccount, btcAccount } = mockETHandBTCAccounts()
+
+    store.dispatch(
+      portfolioSlice.actions.upsertPortfolio(
+        accountToPortfolio({
+          [ethPubKeys[0]]: ethAccount,
+          [btcPubKeys[0]]: btcAccount
+        })
+      )
+    )
+
+    const ethAccountId = `${ethAccount.caip2}:${toLower(ethAccount.pubkey)}`
+    const btcAccountId = `${btcAccount.caip2}:${btcAccount.pubkey}`
```

Comment:
You're not really testing values here so much as testing the logic in the test matches the logic in the store reducer. A hard-coded string value would test the expected output, not the logic.

Reply:
This is just interpolating the values from the given account. I just moved these into the account mock file and they can be exported from there instead of being redefined everywhere.